pilot/pkg/config/kube/ingress/conversion.go

```diff
@@ -266,6 +274,30 @@ func ingressBackendToHTTPRoute(backend *v1beta1.IngressBackend, namespace string
 	}
 }
 
+func resolveNamedPort(backend *v1beta1.IngressBackend, namespace string, client kubernetes.Interface) int32 {
+	svc, err := client.CoreV1().Services(namespace).Get(context.TODO(), backend.ServiceName, metaV1.GetOptions{})
+	if err != nil {
+		return -1
+	}
+	opts := metaV1.ListOptions{
+		LabelSelector: k8sLabels.Set(svc.Spec.Selector).String(),
+	}
+	podList, err := client.CoreV1().Pods("").List(context.TODO(), opts)
```

Comment:
This shouldn't be across all namespaces

Reply:
Done, fetching from same namespace as the svc